transient/configuration.py

```diff
@@ -0,0 +1,297 @@
+"""Supports the creation and validation of Transient-run configurations
+"""
+
+import logging
+import toml
+
+from marshmallow import Schema, fields, post_load, pre_load, ValidationError
+
+from typing import Any, Dict, List, MutableMapping
+
+
+class ConfigFileParsingError(Exception):
+    """Raised when a parsing error is encountered while loading the
+       configuration file
+    """
+
+    pass
+
+
+class ConfigFileOptionError(Exception):
+    """Raised when an invalid configuration option value is encountered in the
+       configuration file
+    """
+
+    pass
+
+
+class CLIArgumentError(Exception):
+    """Raised when an invalid command line argument is encountered
+    """
+
+    pass
+
+
+class Config(Dict[Any, Any]):
+    """Creates an argument dictionary that allows dot notation to access values
+
+    Example:
+
+        >>> args = Config({'arg1': 1, 'arg2': 2})
+        >>> args['arg1'] == args.arg1
+
+    """
+
+    def __getattr__(self, attr: Any) -> Any:
+        return self.get(attr)
+
+    def __setattr__(self, key: Any, value: Any) -> None:
+        self.__setitem__(key, value)
+
+    def __delattr__(self, item: Any) -> None:
+        self.__delitem__(item)
+
+    def __delitem__(self, key: Any) -> None:
+        super(Config, self).__delitem__(key)
+        del self.__dict__[key]
```

Comment:
Same comment as before, why do we need to define `__delitem__`? I assume the default behavior would be fine.